File: April/Abiji-2020/C++/N-th Tribonacci Number/Nth Tribonacci Number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int a=0,b=1,c=1;
        int t;
        if (n<2) return n;
        while(n>2){
            t=a+b+c;
            a = b;
            b = c;
            c = t;
            n--;
        }
        return c;
    }
};

File: April/Abiji-2020/Golang /Nth Tribonacci Number/Nth Tribonacci Number.go
func tribonacci(n int) int {
    var F [38]int
    F[0] = 0
    F[1] = 1
    F[2] = 1
    if n<3{
        return F[n]
    } 
    for i:=3;i<=n;i++{
        F[i] = F[i-1]+F[i-2]+F[i-3]
    }
    return F[n]
}
